static/app/components/searchSyntax/parser.tsx

```diff
@@ -747,6 +767,12 @@ export class TokenConverter {
       return {reason: t('Lists should not have empty values')};
     }
 
+    const hasWildCard = items.some(item => item.value.value.includes('*'));
+
+    if (this.config.disallowWildcard && hasWildCard) {
+      return {reason: t('Lists should not have wildcard values')};
+    }
```

Comment:
Let's inline the `hasWildCard` otherwise when the option is off (which it is for most search bars) we are still doing the work to see if the string has a wildcard.

```suggestion
    if (
      this.config.disallowWildcard && 
      items.some(item => item.value.value.includes('*')
    ) {
      return {reason: t('Lists should not have wildcard values')};
    }
```